app/controllers/reports_controller.rb

```diff
@@ -21,15 +21,15 @@ def edit; end
   def create
     @report = current_user.reports.new(report_params)
 
-    if @report.save
+    if @report.exec_create_transaction
```

Comment:
```suggestion
    if @report.save_with_mentions
```

みたいな名前の方が、挙動がわかりやすいかもしれません。
（メソッドの定義を見に行かなくてもなんとなく何をやっているか想像の付くメソッド名が理想です）